### DISCUSSION TITLE
Common interface for plugins?

### DISCUSSION THREAD

**Discussion by chengenzhao** [gluonhq/scenebuilder#495] [Ideas]
Hi:

Our group(Chinese JavaFX User Group) do have some members have their own projects based on JavaFX
like CSS editor plugin(author already discuss this on Twitter:https://twitter.com/LeeWyatt_7788/status/1485836423446278145)
which was originally designed as Intellij IDEA Plugin and should works great for Scene Builder
and Animated Character(the author has a prototype project here: https://github.com/wkgcass/Chara) etc.
and they would like to contribute these projects to Scene Builder
However integrating all these projects into a single monolithic repo is probably not a good idea
so is it possible for Scene Builder to provide a common plugin interface?
Thus developers could easily expand this project by providing plugins?
I think this function would be also useful for other projects like FXGL?

Thanks

**Comment by AlmasB** [comment]
Yes that's right, custom controls and tools should live outside of "kit" and "app". For the time being, #461 was created as umbrella issue for all JavaFX non-core controls, tools, etc.
